UnitsNet/CompiledLambdas.cs

```diff
@@ -0,0 +1,311 @@
+    /// Compiled lambda expressions that can be invoked with generic run-time parameters.
+    /// </summary>
+    internal static class CompiledLambdas
+    {
+        /// <summary>
+        /// Multiplies the given values.
+        /// </summary>
+        /// <typeparam name="T">The type of the operation (left hand side, right hand side, and result).</typeparam>
+        /// <param name="left">The left hand side parameter.</param>
+        /// <param name="right">The right hand side parameter.</param>
+        /// <returns>The multiplied result.</returns>
+        internal static T Multiply<T>(T left, T right) => Multiply<T, T, T>(left, right);
+
+        /// <summary>
+        /// Multiplies the given values.
+        /// </summary>
+        /// <typeparam name="TLeft">The type of the left hand side.</typeparam>
+        /// <typeparam name="TRight">The type of the right hand side.</typeparam>
+        /// <typeparam name="TResult">The result type.</typeparam>
+        /// <param name="left">The left hand side parameter.</param>
+        /// <param name="right">The right hand side parameter.</param>
+        /// <returns>The multiplied result.</returns>
+        internal static TResult Multiply<TLeft, TRight, TResult>(TLeft left, TRight right) =>
+            MultiplyImplementation<TLeft, TRight, TResult>.Invoke(left, right);
+
+        /// <summary>
+        /// Divides the given values.
+        /// </summary>
+        /// <typeparam name="T">The type of the operation (left hand side, right hand side, and result).</typeparam>
+        /// <param name="left">The left hand side parameter.</param>
+        /// <param name="right">The right hand side parameter.</param>
+        /// <returns>The divided result.</returns>
+        internal static T Divide<T>(T left, T right) => Divide<T, T, T>(left, right);
+
+        /// <summary>
+        /// Divides the given values.
+        /// </summary>
+        /// <typeparam name="TLeft">The type of the left hand side.</typeparam>
+        /// <typeparam name="TRight">The type of the right hand side.</typeparam>
+        /// <typeparam name="TResult">The result type.</typeparam>
+        /// <param name="left">The left hand side parameter.</param>
+        /// <param name="right">The right hand side parameter.</param>
+        /// <returns>The divided result.</returns>
+        internal static TResult Divide<TLeft, TRight, TResult>(TLeft left, TRight right) =>
+            DivideImplementation<TLeft, TRight, TResult>.Invoke(left, right);
+
+        /// <summary>
+        /// Adds the given values.
+        /// </summary>
+        /// <typeparam name="T">The type of the operation (left hand side, right hand side, and result).</typeparam>
+        /// <param name="left">The left hand side parameter.</param>
+        /// <param name="right">The right hand side parameter.</param>
+        /// <returns>The added result.</returns>
+        internal static T Add<T>(T left, T right) => Add<T, T, T>(left, right);
```

Comment:
Any reason to not call `AddImplementation` directly here? Same for other places. 
The JIT probably inlines this, but I don't know if it does.

Reply:
I would think so, but I called them directly instead